pkg/platform/kube/platform.go

```diff
@@ -103,6 +105,13 @@ func (p *Platform) CreateFunction(createFunctionOptions *platform.CreateFunction
 	var existingFunctionInstance *nuclioio.NuclioFunction
 	var existingFunctionConfig *functionconfig.ConfigWithStatus
 
+	// validate function name is according to k8s convention
```

Comment:
not the correct place, please add check to https://github.com/nuclio/nuclio/blob/master/pkg/platform/kube/controller/nucliofunction.go#L85